src/modules/cmdpal/ext/Microsoft.CmdPal.Ext.Bookmark/Helpers/CommandLineHelper.cs

```diff
@@ -0,0 +1,294 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.CmdPal.Ext.Bookmarks.Helpers;
+
+/// <summary>
+/// Provides helper methods for parsing command lines and expanding paths.
+/// </summary>
+/// <remarks>
+/// Warning: This code handles parsing specifically for Bookmarks, and is NOT a general-purpose command line parser.
+/// In some cases it mimics system rules (e.g. CreateProcess, CommandLineToArgvW) but in other cases it uses, but it can also
+/// bend the rules to be more forgiving.
+/// </remarks>
+internal static partial class CommandLineHelper
```

Comment:
note to self: in 0.7, we'll probably want to dedupe this file with the logic in the run command provider, and handle both the same way as this file